ledger/participant-integration-api/src/test/suite/scala/platform/store/dao/BufferedTransactionByIdReaderSpec.scala

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.dao
+
+import com.daml.ledger.offset.Offset
+import com.daml.lf.data.Ref
+import com.daml.lf.data.Ref.Party
+import com.daml.lf.data.Time.Timestamp
+import com.daml.logging.LoggingContext
+import com.daml.platform.store.cache.InMemoryFanoutBuffer
+import com.daml.platform.store.dao.BufferedTransactionByIdReader.{
+  FetchTransactionByIdFromPersistence,
+  ToApiResponse,
+}
+import com.daml.platform.store.interfaces.TransactionLogUpdate
+import org.mockito.MockitoSugar
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.Future
+
+class BufferedTransactionByIdReaderSpec extends AsyncFlatSpec with MockitoSugar with Matchers {
+  private val className = classOf[BufferedTransactionByIdReader[_]].getSimpleName
+
+  private implicit val loggingContext: LoggingContext = LoggingContext.ForTesting
+
+  private val requestingParties = Set("p1", "p2").map(Ref.Party.assertFromString)
+
+  private val bufferedTransactionId1 = "bufferedTid_1"
+  private val bufferedTransactionId2 = "bufferedTid_2"
+  private val notBufferedTransactionId = "notBufferedTid"
+  private val unknownTransactionId = "unknownTransactionId"
+
+  private val bufferedTransaction1 = tx(bufferedTransactionId1)
+  private val bufferedTransaction2 = tx(bufferedTransactionId2)
+
+  private val inMemoryFanout = mock[InMemoryFanoutBuffer]
```

Comment:
It may have been just as easy to use the real implementation here.

Reply:
yes I could, but then I would need to expose some details to the test that we don't care about (e.g. the offset to be used when pushing to the buffer, the maxBufferChunkSize etc.). I would leave it like this since it's a bit clearer.